cpp/arcticdb/pipeline/write_frame.cpp

```diff
@@ -83,20 +83,26 @@ folly::Future<std::vector<SliceAndKey>> write_slices(
             auto rows_to_write = slice.row_range.second - slice.row_range.first;
             if (frame.desc.index().field_count() > 0) {
                 util::check(static_cast<bool>(frame.index_tensor), "Got null index tensor in write_slices");
-                aggregator_set_data(
+                auto opt_error = aggregator_set_data(
                     frame.desc.fields(0).type(),
                     frame.index_tensor.value(),
                     agg, 0, rows_to_write, offset_in_frame, slice_num_for_column, regular_slice_size, false);
+                if (opt_error.has_value()) {
```

Comment:
Normally we just use the implicit bool conversion.

Reply:
I don't think we're very consistent on this:
```
(dev) [root@d22c3e3750f4 arcticdb]# grep -rI "has_value()" | wc -l
172
```

Reply:
Fair enough, I'm not that fussed. has_value is arguably clearer anyway